full-stack-practice/server/server.ts

```diff
@@ -42,22 +42,15 @@ app.get('/api/products/:productId', async (req, res, next) => {
     if (!Number.isInteger(productId) || productId < 1) {
       throw new ClientError(400, 'productId must be a positive integer');
     }
-
     const sql = `
       SELECT *
         FROM "products"
         WHERE "productId" = $1;
     `;
-    const params = [productId];
-    const result = await db.query<Product>(sql, params);
-
+    const result = await db.query<Product>(sql, [productId]);
     if (result.rows.length === 0) {
-      // No product found
-      return res
-        .status(404)
-        .json({ error: `Product with productId ${productId} not found` });
+      throw new ClientError(400, 'productId must be a positive integer');
```

Comment:
Oops, copy/paste error! You should have changed the status code to 404 and the message to say product not found.